#### COMMON STATES FOR ALL STATE MACHINES ####
int8 COLLISION = -1
int8 OFF_ROAD = -2
int8 FORBIDDEN_LEFT = -3


##### PROGRESS STATES #####
# These states enable high level information about the drive (e.g. not left start zone, reached end zone)
int8 PROGRESS_BEFORE_START = 10
int8 PROGRESS_RUNNING = 11
int8 PROGRESS_FINISHED = 12


##### OVERTAKING STATES #####
int8 OVERTAKING_BEFORE_START = 20
int8 OVERTAKING_RIGHT = 21
int8 OVERTAKING_LEFT = 22


##### PARKING STATES #####
int8 PARKING_BEFORE_START = 30
int8 PARKING_IN_ZONE = 31
int8 PARKING_ATTEMPT = 32
int8 PARKING_IN_SPOT = 33
int8 PARKING_SUCCESS = 34
int8 PARKING_OUT = 35
#int8 PARKING_AFTER_RIGHT = 36

int8 PARKING_FORBIDDEN_RIGHT = -30
int8 PARKING_FORBIDDEN_LEFT = -31
int8 PARKING_FORBIDDEN_IN_LOT = -32


##### PRIORITY STATES #####
int8 PRIORITY_BEFORE_START = 40
int8 PRIORITY_IN_STOP_ZONE = 41
int8 PRIORITY_IN_HALT_ZONE = 42
int8 PRIORITY_SUCCESS_STOP = 43

int8 PRIORITY_FORBIDDEN_IN_STOP_ZONE = -40

#---- Msg ----- #
int8 state
